src/shims/time.rs

```diff
@@ -177,4 +178,40 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.write_packed_immediates(info, &imms)?;
         Ok(0) // KERN_SUCCESS
     }
+
+    fn nanosleep(
+        &mut self,
+        req_op: OpTy<'tcx, Tag>,
+        _rem: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i32> {
+        // Signal handlers are not supported, so rem will never be written to.
+
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("nanosleep")?;
+
+        let duration = match this.read_timespec(req_op)? {
+            Ok(duration) => duration,
+            Err(TimespecError) => {
+                let einval = this.eval_libc("EINVAL")?;
+                this.set_last_error(einval)?;
+                return Ok(-1);
+            }
+        };
+        let timeout_time = Time::RealTime(SystemTime::now().checked_add(duration).unwrap());
```

Comment:
Wouldn't `Time::Monotonic` be more appropriate here? I assume a `nanosleep` does not wake up early when the system clock is moved backwards.